code/modules/mob/living/basic/drone/_drone.dm

```diff
@@ -169,19 +168,16 @@
 		/obj/machinery/atmospherics/components/unary/vent_scrubber,
 	)
 
-/mob/living/simple_animal/drone/Initialize(mapload)
+/mob/living/basic/drone/Initialize(mapload)
 	. = ..()
 	GLOB.drones_list += src
-	access_card = new /obj/item/card/id/advanced/simple_bot(src)
+	AddElement(/datum/element/dextrous, hud_type = hud_type)
 	AddComponent(/datum/component/basic_inhands, y_offset = getItemPixelShiftY())
-
-	// Doing this hurts my soul, but simple_animal access reworks are for another day.
-	var/datum/id_trim/job/cap_trim = SSid_access.trim_singletons_by_path[/datum/id_trim/job/captain]
-	access_card.add_access(cap_trim.access + cap_trim.wildcard_access)
+	AddComponent(/datum/component/simple_access, SSid_access.get_region_access_list(list(REGION_ALL_GLOBAL)), src)
```

Comment:
The `donor_atom` argument is optional and seems basically meaningless if you are passing it as `src`.
It seems like its purpose is to allow the access to be removed if some _other_ object is destroyed.